Valigator.Text.Json/ValigatorJsonPropertyHandler.cs

```diff
@@ -18,12 +18,84 @@ internal abstract class ValigatorJsonPropertyHandler<TObject>
 
 		public static ValigatorJsonPropertyHandler<TObject> Create(PropertyInfo property)
 		{
-			var handlerType = typeof(ValigatorJsonPropertyHandler<,>).MakeGenericType(typeof(TObject), property.PropertyType.GetGenericArguments()[0]);
+			var propertyHandlerType = property.PropertyType.IsValigatorDataType() ? typeof(ValigatorJsonPropertyHandler<,>) : typeof(NonValigatorDataJsonPropertyHandler<,>);
+
+			var propertyType = property.PropertyType.IsValigatorDataType() ? property.PropertyType.GetGenericArguments()[0] : property.PropertyType;
+
+			var handlerType = propertyHandlerType.MakeGenericType(typeof(TObject), propertyType);
 
 			var createMethod = handlerType.GetMethod(nameof(ValigatorJsonPropertyHandler<TObject, bool>.Create), BindingFlags.Public | BindingFlags.Static, Type.DefaultBinder, new[] { typeof(PropertyInfo) }, null);
 
 			return (ValigatorJsonPropertyHandler<TObject>)createMethod.Invoke(null, new[] { property });
 		}
+
+		internal static Func<TObject, TValue> CreateGetter<TValue>(PropertyInfo property)
+		{
+			var objParameter = Expression.Parameter(typeof(TObject), "obj");
+			var propertyParameter = Expression.Constant(property);
+
+			var getValueMethod = property.GetType().GetMethod(nameof(PropertyInfo.GetValue), new[] { typeof(object) });
+			var getterExpression = Expression.Convert(Expression.Call(propertyParameter, getValueMethod, objParameter), property.PropertyType);
+			var getter = property.CanRead ? Expression.Lambda<Func<TObject, TValue>>(getterExpression, objParameter).Compile() : null;
```

Comment:
Is there a way to avoid the null? or prvoide a lambda that throws a specific exception?